DeviceAdministration/Infrastructure.UnitTests/Simulator.WebJob/AppConfigRepositoryTests.cs

```diff
@@ -0,0 +1,108 @@
+﻿using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Common.Configurations;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Common.Models;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Simulator.WebJob.SimulatorCore.Logging;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Simulator.WebJob.SimulatorCore.Repository;
+using Microsoft.Azure.Devices.Applications.RemoteMonitoring.Simulator.WebJob.SimulatorCore.Telemetry;
+using Moq;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Azure.Devices.Applications.RemoteMonitoring.DeviceAdmin.Infrastructure.UnitTests.Simulator.WebJob
+{
+    public class AppConfigRepositoryTests
+    {
+        private readonly Mock<ILogger> _loggerMock;
+        private readonly Mock<IConfigurationProvider> _configurationProviderMock;
+        private AppConfigRepository repo;
+
+        private string testHostName;
+
+        public AppConfigRepositoryTests()
+        {
+            this._loggerMock = new Mock<ILogger>();
+            this._configurationProviderMock = new Mock<IConfigurationProvider>();
+            this.testHostName = "testHostName";
+
+            this._configurationProviderMock.Setup(mock => mock.GetConfigurationSettingValue("iotHub.HostName")).Returns(this.testHostName);
+
+            this.repo = new AppConfigRepository(this._configurationProviderMock.Object, this._loggerMock.Object);
+        }
+
+        [Fact]
+        public async void GetDeviceListAsyncTest()
+        {
+            var deviceList = await this.repo.GetDeviceListAsync();
+
+            Assert.NotEmpty(deviceList);
+            Assert.Equal(deviceList[0].HostName, this.testHostName);
+        }
+
+        [Fact]
+        public async void GetDeviceAsyncTest()
+        {
+            Assert.Null(await this.repo.GetDeviceAsync(""));
+
+            var deviceList = await this.repo.GetDeviceListAsync();
```

Comment:
you have to call ```GetDeviceListAsync()``` to first populate the list...otherwise all other calls will just return early or their false case...and if we use a random generated id it will always fail unless we add it first...wanted to limit how many other functions I called while testing this one